src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -0,0 +1,624 @@
+                const parameters = [
+                    createParameter('connectorName', response.data.connectorName),
+                    createParameter('pluginName', response.data.pluginName)
+                ];
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.ERROR, 'query.editor.inactive.plugin.warning.msg', parameters);
+            };
+
+            const createConnectorProgressDialog = (actionName, iri, connectorName) => {
+                const progressScope = $scope.$new(true);
+                progressScope.beingBuiltConnector = {
+                    percentDone: 0,
+                    status: {
+                        processedEntities: 0,
+                        estimatedEntities: 0,
+                        indexedEntities: 0,
+                        entitiesPerSecond: 0
+                    },
+                    actionName,
+                    eta: "-",
+                    inline: false,
+                    iri,
+                    name: connectorName,
+                    doneCallback: function () {
+                        connectorProgressModal.dismiss('cancel');
+                    }
+                };
+                progressScope.getHumanReadableSeconds = $scope.getHumanReadableSeconds;
+
+                const connectorProgressModal = $uibModal.open({
+                    templateUrl: 'pages/connectorProgress.html',
+                    controller: 'CreateProgressCtrl',
+                    size: 'lg',
+                    backdrop: 'static',
+                    scope: progressScope
+                });
+                return connectorProgressModal;
+            };
+
+            const toCreateCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.creating'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'created.connector', getCommandParameters(response));
+            };
+
+            const toRepairCommandResponse = (response, tabId) => {
+                const connectorProgressModal = createConnectorProgressDialog($translate.instant('externalsync.repairing'), response.data.iri, response.data.name);
+                tabIdToConnectorProgressModalMapping.set(tabId, connectorProgressModal);
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'query.editor.repaired.connector', getCommandParameters(response));
+            };
+
+            const toDropCommandResponse = (response) => {
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS, 'externalsync.delete.success.msg', getCommandParameters(response));
+            };
+
+            const getBeforeUpdateQueryHandler = () => (query, tabId) => {
```

Comment:
I think that this function can be simplified. Try removing the `new Promise...` thing and put `Promise.resolve(...)` inline where you have these `resolve` calls. This would remove one level of nesting.

Reply:
You can even remove the `resolve`-ers from the first `then` altogether I think and just do `return whatever_command_you_want()`, then if you need to do the resolve (which I think you don't) you can chain another then  and resolve there, like ```.then((command) => return Promise.resolve(command))```. I havent tested it though, but it should work and it'd simplify it even greater.

Reply:
Done